plugins/org.eclipse.n4js.xpect.ui/src/org/eclipse/n4js/xpect/ui/common/XpectN4JSES5TranspilerHelper.java

```diff
@@ -134,19 +133,22 @@ public String doCompileAndExecute(final XtextResource resource,
 			// If we are in the IDE, execute the test the same as for "Run in Node.js" and this way avoid
 			// the effort of calculating dependencies etc.
 
-			final String implementationId = chooseImplHelper.chooseImplementationIfRequired(NodeRunner.ID,
-					resource.getURI().trimFileExtension());
+			Path packagesPath = createTemporaryYarnWorkspace(artificialRoot);
 
 			boolean replaceQuotes = false;
 			// We have to generate JS code for the resource. Because if Xpect test is quickfixAndRun the resource
 			// contains errors and hence no generated JS code is available for execution.
 			// Then sneak in the path to the generated JS code.
 			Script script = (Script) resource.getContents().get(0);
-			createTempJsFileWithScript(artificialRoot.toPath(), script, options, replaceQuotes);
-			runConfig = runnerFrontEnd.createConfiguration(NodeRunner.ID,
-					(implementationId == ChooseImplementationHelper.CANCEL) ? null : implementationId,
-					systemLoader.getId(),
-					resource.getURI().trimFileExtension(), artificialRoot.getAbsolutePath().toString());
+			createTempJsFileWithScript(packagesPath, script, options, replaceQuotes).toPath().getParent();
+
+			String fileToRun = jsModulePathToRun(script);
+			fileToRun = fileToRun.substring(fileToRun.indexOf('/') + 1);
```

Comment:
What about(TM) project names of scoped projects such as `@n4jsd/lodash`? (Mind other occurrence below)